doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java

```diff
@@ -150,6 +145,13 @@ public void updateDoiRequest(UUID publicationID, DoiRequestStatus requestedStatu
         putItem(publication);
     }
 
+    private Publication fetchPublicationForUser(CreateDoiRequest createDoiRequest, String username)
+        throws NotFoundException, ForbiddenException {
+        var publication = fetchPublication(UUID.fromString(createDoiRequest.getPublicationId()));
```

Comment:
Worth noting that CreateDoiRequest::getPublicationId always gets called wrapped with UUID.fromString in usage contexts. Probably means this field has the type UUID. The handling of IllegalArgumentExceptions should probably be moved to commons parseBody as this is generic functionality.